File: app/src/main/java/com/pineapple/create_tms_activity/Task.java
package com.pineapple.create_tms_activity;

public class Task {
    public String subName;
    public String time;
    public String textName;
    public String page;
    public String date;

    public Task(){
    }

    public Task(String subName,String time,String textName,String page,String date){
        this.subName = subName;
        this.subName = time;
        this.textName = textName;
        this.page = page;
        this.date = date;
    }

    public String getSubName() {
        return subName;
    }

    public void setSubName(String subName) {
        this.subName = subName;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getTextName() {
        return textName;
    }

    public void setTextName(String textName) {
        this.textName = textName;
    }

    public String getPage() {
        return page;
    }

    public void setPage(String page) {
        this.page = page;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }
}

File: app/src/main/java/com/pineapple/create_tms_activity/Task.kt
package com.pineapple.create_tms_activity

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import java.util.*

// TODO: data class から open classに変えよう
// TODO: primary key を追加しよう
// @PrimaryKey
// open var id: String = UUID.randomUUID().toString()
open class Task (
    @PrimaryKey
    open var id: String = UUID.randomUUID().toString(),
    open val subName: String,
    open val time: String,
    open val textName: String,
    open val page: String,
    open val date: String

) : RealmObject()

// TODO: RealmObjectを継承しよう

// https://qiita.com/souhei-etou/items/d8a30626ab2c300e4427